app/controllers/v1/users_controller.rb

```diff
@@ -27,12 +27,14 @@ def create
   end
 
   def update
-    @user = User.find_by_username(params[:username])
-
-    if @user.update(user_params)
+    @user = User.find_by(id: params[:id])
+    authorize @user
+    if @user.update(permitted_attributes(@user)) && permitted_attributes(@user) != {}
```

Comment:
If you change order, you won't event start update if attributes hash is empty.
`permitted_attributes(@user) != {} && @user.update(permitted_attributes(@user))`

Reply:
Can you clarify what you mean by that? :)